### PR TITLE
fix(gen): use mapping key for discriminator const

### PR DESCRIPTION
**Pull request by Liooo**
Closes: https://github.com/ogen-go/ogen/issues/967

Just changing the generated string constant
This won't allow us to use the same value as was stated in https://github.com/ogen-go/ogen/issues/967#issue-1787529410.

Please tell me if any changes to be made are missing.

### CODE REVIEW

**Comment by Liooo** [comment]
I ran go generate and noticed there's more to be done. making this draft.

**Comment by Liooo** [comment]
@tdakkota 

Please review, at least CI is passing.

**Review by tdakkota — CHANGES_REQUESTED** [review]
One small change in a template.

Everything else is fine.  Thank you!

File: gen/_template/schema/sum.tmpl [hunk 1]
```diff
@@ -14,7 +14,14 @@ type {{ $.Name }}Type string
 // Possible values for {{ $.Name }}Type.
 const (
 	{{- range $s := $.SumOf }}
-		{{ $s.Name }}{{ $.Name }} {{ $.Name }}Type = {{ quote $s.Go }}
+	{{- $m := $.SumSpec.PickMappingEntryFor $s }}
+	{{- if ne $m nil }}
```
[the remark this answers is not in the document]
Comment (Liooo) [reply gen/_template/schema/sum.tmpl]: @tdakkota thank you

**Review by tdakkota — APPROVED** [review]

**Pull request merged**